### PR TITLE
Improve the HMAC code.

### PR DESCRIPTION
**Pull request by haydenroche5**
- Add a get private key method.
- Add a public key comparison method.

Adding these resolves some failing OpenSSL unit tests when testing with 1.1.1m.

### CODE REVIEW

File: src/we_mac.c [hunk 1]
```diff
@@ -1331,12 +1331,87 @@ static int we_hmac_set_priv_key(EVP_PKEY *pkey, const unsigned char *priv,
...
+{
+    int ret = 1;
+    ASN1_OCTET_STRING* aString;
+    ASN1_OCTET_STRING* bString;
+
+    WOLFENGINE_ENTER(WE_LOG_MAC, "we_hmac_pkey_pub_cmp");
+
+    if (a == NULL || b == NULL) {
+        WOLFENGINE_ERROR_MSG(WE_LOG_MAC, "Bad argument.");
+        ret = 0;
```
Comment (SparkiDev) [inline src/we_mac.c:1395]: -2 on error?

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/we_mac.c (hunk 1 above)
Comment (haydenroche5) [reply src/we_mac.c:1395]: Is this a suggestion in order to distinguish mismatches from errors, or is -2 on error documented somewhere? I'm good with the change either way, just curious.

Comment (SparkiDev) [reply src/we_mac.c:1395]: I checked OpenSSL and they are returning -2 for this type of error in related functions: eckey_pub_cmp().

Comment (haydenroche5) [reply src/we_mac.c:1395]: The OpenSSL code is in hm_ameth.c:

```
static int hmac_pkey_public_cmp(const EVP_PKEY *a, const EVP_PKEY *b)
{
    return ASN1_OCTET_STRING_cmp(EVP_PKEY_get0(a), EVP_PKEY_get0(b));
}
```

It doesn't bother checking the arguments.

Comment (SparkiDev) [reply src/we_mac.c:1395]: In theory, the parameters will always be good as the caller will have validated them.
But, for future proofing, we should check.

EVP_PKEY_cmp returns the result from the call to a->ameth->pub_cmp().
Previously in the function it returns -1 when types don't match and -2 when no ameth on a.

Comment (haydenroche5) [reply src/we_mac.c]: Modified it to return -2 on error.

**Pull request merged**